### ISSUE TITLE
Use ListClusters/DescribeCluster instead of DescribeJobFlows

### ISSUE THREAD

**Issue by bchess** [Yelp/mrjob#876]
There are new APIs for describing EMR state. They're apparently much faster:

https://forums.aws.amazon.com/ann.jspa?annID=2213#

Consider using these new APIs in place of DescribeJobFlows

**Comment by davidmarin** [comment]
May need to do this in v0.4.5; looks like EMR is now giving 400 Bad Request for using this API call. See [this thread](https://github.com/Yelp/mrjob/issues/876) on the Google Group. Going to try contacting the EMR team and telling them not to deprecate this yet.

**Comment by davidmarin** [comment]
Looks like these API calls were added to boto in 2.16.0.

**Comment by davidmarin** [comment]
The calls don't actually work correctly for `statechangereason` until 2.35.0.

**Comment by davidmarin** [comment]
Using a shim with code form 2.35.0, plus a fix for step status (see https://github.com/boto/boto/issues/3268)

**Comment by davidmarin** [comment]
The new APIs apparently don't give cluster "start" time any more (the time the cluster began bootstrapping), so that leads to a few subtle changes:
 * clock for time-to-end-of-hour starts at cluster creation time
 * jobs stuck in the STARTING state are considered long-running, not idle

**Comment by davidmarin** [comment]
Whoops:

```
Traceback (most recent call last):
  File "terminate_idle_job_flows.py", line 427, in <module>
    main()
  File "terminate_idle_job_flows.py", line 112, in main
    **runner_kwargs(options)
  File "terminate_idle_job_flows.py", line 183, in maybe_terminate_clusters
    if is_cluster_non_streaming(steps):
  File "terminate_idle_job_flows.py", line 274, in is_cluster_non_streaming
    if DEBUG_JAR_RE.match(arg):
```

I'm guessing the actual args are in the `value` field of an `EmrObject`. Looks like I based `mockboto` too heavily on the EMR API docs and not heavily enough on actual boto behavior.

**Comment by davidmarin** [comment]
Also:

```
Traceback (most recent call last):
  File "job_flow_pool.py", line 29, in <module>
    from mrjob.emr import est_time_to_hour
ImportError: cannot import name est_time_to_hour
```

**Comment by davidmarin** [comment]
Fixed by #1098
